inlong-sort/sort-single-tenant/src/main/java/org/apache/inlong/sort/singletenant/flink/parser/impl/FlinkSqlParser.java

```diff
@@ -0,0 +1,379 @@
+        });
+        for (FieldInfo field : loadNode.getFields()) {
+            FieldRelationShip fieldRelation = fieldRelationMap.get(field.getName());
+            if (fieldRelation != null) {
+                sb.append(fieldRelation.getInputField().format()).append(" AS ").append(field.format()).append(",");
+            } else {
+                String targetType = TableFormatUtils.deriveLogicalType(field.getFormatInfo()).asSerializableString();
+                sb.append("CAST(NULL as ").append(targetType).append(") AS ").append(field.format()).append(",");
+            }
+        }
+        sb.deleteCharAt(sb.length() - 1);
+        sb.append(" FROM `").append(inputNode.genTableName()).append("`");
+        if (loadNode.getFilters() != null && !loadNode.getFilters().isEmpty()) {
+            sb.append(" WHERE");
+            for (FilterFunction filter : loadNode.getFilters()) {
+                sb.append(" ").append(filter.format());
+            }
+        }
+        return sb.toString();
+    }
+
+    private String genCreateSql(Node node) {
+        if (node instanceof TransformNode) {
+            return genCreateTransformSql(node);
+        }
+        StringBuilder sb = new StringBuilder("CREATE TABLE `");
+        sb.append(node.genTableName()).append("`(\n");
+        sb.append(genPrimaryKey(node.getPrimaryKey()));
+        sb.append(parseFields(node.getFields(), node instanceof LoadNode));
+        if (node instanceof ExtractNode) {
+            ExtractNode extractNode = (ExtractNode) node;
+            if (extractNode.getWatermarkField() != null) {
+                sb.append(",\n     ").append(extractNode.getWatermarkField().format());
+            }
+        }
+        sb.append(")");
+        if (node.getPartitionFields() != null && !node.getPartitionFields().isEmpty()) {
+            sb.append(String.format("\nPARTITIONED BY (%s)",
+                    StringUtils.joinWith(",", formatFields(node.getPartitionFields()))));
+        }
+        sb.append(parseOptions(node.tableOptions()));
+        return sb.toString();
+    }
+
+    private String genCreateTransformSql(Node node) {
+        return String.format("CREATE VIEW `%s` (\n%s)",
+                node.genTableName(), parseTransformNodeFields(node.getFields()));
+    }
+
+    private String parseOptions(Map<String, String> options) {
+        StringBuilder sb = new StringBuilder();
+        if (options != null && !options.isEmpty()) {
+            sb.append("\n    WITH (");
+            for (Map.Entry<String, String> kv : options.entrySet()) {
+                sb.append("\n    '").append(kv.getKey()).append("' = '").append(kv.getValue()).append("'").append(",");
+            }
+            if (sb.length() > 0) {
+                sb.delete(sb.lastIndexOf(","), sb.length());
+            }
+            sb.append("\n)");
+        }
+        return sb.toString();
+    }
+
+    public String parseTransformNodeFields(List<FieldInfo> fields) {
```

Comment:
This method was `public`, does it necessary?

Reply:
It is not necessary, I'll deal with it later.

Reply:
done